### ISSUE TITLE
Add Campaign Fields API

### ISSUE THREAD

**Issue by ericnicolaas** [Charitable/Charitable#530]
In 1.5.0 we introduced the Donation Fields API. See https://github.com/Charitable/Charitable/issues/402. In 1.6.0, I want to add a Campaign Fields API.

This should be modelled on the Donation Fields API. 

**Register a field**
```
// Define a new field.
$field = new Charitable_Campaign_Field( 'gallery', array(
    'label' => __( 'Gallery', 'charitable' ),
    'data_type' => 'user',
    'ambassadors_form' => array( /* various args */ ),
    'admin_form' => array( /* various args */ ),
    'show_in_export' => true,
    'email_tag' => array(
        'description' => __( 'The campaign gallery' , 'charitable' ),
    ) )
) );

// Register the field.
charitable()->campaign_fields()->register( $field );
```

When a field is registered, it can be automatically added to:

- The campaign settings meta box (https://github.com/Charitable/Charitable/issues/531)
- Ambassadors campaign form
- Campaign export (https://github.com/Charitable/Charitable/issues/529)
- Email tag

**Comment by helgatheviking** [comment]
How soon does Recurring need this?

**Comment by ericnicolaas** [comment]
@helgatheviking I'm designing it to work in a backwards-compatible way. So it should **just work** with Recurring the way it is now. Lots of extensions will be in the same place as Recurring, so I fully expect it to take a while to transition everything across.

**Comment by helgatheviking** [comment]
ok good, b/c i don't think i can handle more stuff! :)

**Comment by ericnicolaas** [comment]
Closed by 878e5bbf
